src/main/java/network/brightspots/rcv/CandidateRankingsList.java

```diff
@@ -23,31 +23,19 @@
 import javafx.util.Pair;
 
 class CandidateRankingsList implements Iterable<Pair<Integer, CandidatesAtRanking>> {
-  class CandidateRankingsListIterator implements Iterator<Pair<Integer, CandidatesAtRanking>> {
-    private int iteratorIndex = 0;
-
-    public boolean hasNext() {
-      return iteratorIndex < rankings.length;
-    }
-
-    public Pair<Integer, CandidatesAtRanking> next() {
-      if (iteratorIndex == rankings.length) {
-        throw new NoSuchElementException();
-      }
-
-      do {
-        iteratorIndex++;
-        assert iteratorIndex <= rankings.length;
-      } while (!hasRankingAt(iteratorIndex));
-
-      // Note: round numbers are 1-indexed externally, 0-indexed internally,
-      // thus why we return a different value than what we index into here
-      return new Pair<>(iteratorIndex, rankings[iteratorIndex - 1]);
-    }
-
-    public void remove() {
-      throw new UnsupportedOperationException();
+  /**
+   * Precondition: must be called with a 1-indexed ranking number. Note: just because this returns
+   * false at ranking N doesn't mean it will also return false at N+1 -- specifically, that will not
+   * be true if there are skipped rankings.
+   *
+   * @param num A value >= 1.
+   * @return Whether the candidate has a ranking at the given value.
+   */
+  boolean hasRankingAt(int num) {
+    if (num < 1) {
+      throw new IllegalArgumentException();
```

Comment:
Curious what motivated this move?

Reply:
Ah, that was automatic IntelliJ stuff. The original three collaborators worked out years ago that we'd keep all the boxes checked when committing in IntelliJ:

![image](https://github.com/BrightSpots/rcv/assets/8259446/31da4891-cd8e-49da-9160-8c88f2a9d88f)

Reply:
Ah, perhaps worth adding this to the contributor guide? https://github.com/BrightSpots/rcv/blob/develop/contributor_guide.md

Reply:
Good idea; created #699 for this.